src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java

```diff
@@ -275,55 +281,79 @@ public void shutdown() {
-        }
-
-        final int tumorNonRef = countNonRef(refBase, tumorContext);
-        if (tumorNonRef < MTAC.minVariantsInPileup) {
-            return new ActivityProfileState(refInterval, 0.0);
-        }
+        final ReadPileup pileup = context.getBasePileup();
+        final ReadPileup tumorPileup = pileup.getPileupForSample(MTAC.tumorSampleName, header);
+        final Pair<Integer, Double> tumorAltCountAndQualSum = altCountAndQualSum(tumorPileup, refBase);
+        final int tumorAltCount = tumorAltCountAndQualSum.getFirst();
+        final int tumorRefCount = tumorPileup.size() - tumorAltCount;
 
-        // since errors are rare, the number of errors (if reads are independent) is approximately a Poisson random variable,
-        // with mean equal to its variance
-        final double expectedTumorNonRefDueToError = StreamSupport.stream(tumorContext.getBasePileup().spliterator(), false)
-                .mapToDouble(pe -> QualityUtils.qualToErrorProb(pe.getQual()))
-                .sum();
-        final double tumorNonRefStdev = Math.sqrt(expectedTumorNonRefDueToError);
+        final double tumorLog10Odds = -QualityUtils.qualToErrorProbLog10(tumorAltCountAndQualSum.getSecond()) +
+                MathUtils.log10Factorial(tumorAltCount) + MathUtils.log10Factorial(tumorRefCount) - MathUtils.log10Factorial(tumorPileup.size() + 1);
 
-        if (tumorNonRef < expectedTumorNonRefDueToError + MTAC.tumorStandardDeviationsThreshold * tumorNonRefStdev) {
+        if (tumorLog10Odds < MTAC.initialTumorLodThreshold) {
             return new ActivityProfileState(refInterval, 0.0);
+        } else if (hasNormal()) {
+            final ReadPileup normalPileup = pileup.getPileupForSample(MTAC.normalSampleName, header);
+            final Pair<Integer, Double> normalAltCountAndQualSum = altCountAndQualSum(normalPileup, refBase);
+            final int normalAltCount = normalAltCountAndQualSum.getFirst();
+            final double normalQualSum = normalAltCountAndQualSum.getSecond();
+            if (normalAltCount > normalPileup.size() * MAX_ALT_FRACTION_IN_NORMAL && normalQualSum > MAX_NORMAL_QUAL_SUM) {
+                return new ActivityProfileState(refInterval, 0.0);
+            }
+        } else {
+            final List<VariantContext> germline = featureContext.getValues(MTAC.germlineResource, refInterval);
+            if (!germline.isEmpty() && germline.get(0).getAttributeAsDoubleList(VCFConstants.ALLELE_FREQUENCY_KEY, 0.0).get(0) > MTAC.maxPopulationAlleleFrequency) {
+                return new ActivityProfileState(refInterval, 0.0);
+            }
         }
 
-        if (hasNormal() && normalContext != null && countNonRef(refBase, normalContext) > normalContext.getBasePileup().size() * MTAC.minNormalVariantFraction) {
+        if (!MTAC.genotypePonSites && !featureContext.getValues(MTAC.pon, new SimpleInterval(context.getContig(), (int) context.getPosition(), (int) context.getPosition())).isEmpty()) {
```

Comment:
Hi, I understood this excludes variants at PON sites entirely and save computing resources. Consequently, variants falling into this will not be part of Mutect output vcf (correct?).
If so, this seems like an inconsistent behavior because there are other variants at PON sites that are part of Mutect output and explicitly filtered as panel_of_normals as opposed to be excluded silently. 
I wonder if the default value of --genotypePonSites would better be true to be consistent and explicit.